akka-stream/src/main/scala/akka/stream/impl/Stages.scala

```diff
@@ -147,6 +151,28 @@ import akka.stream.Attributes._
 
     val inputBoundary = name("input-boundary")
     val outputBoundary = name("output-boundary")
+
+    private val cache = new ConcurrentHashMap[(String, Class[_]), String]
+    private val verboseOperatorNames =
+      // unfortunately this must be a static setting determined from the classpath instead of from the ActorSystem
+      // as there is no ActorSystem necessary to create operators.
+      ConfigFactory.load().getBoolean("akka.stream.verbose-operator-names")
```

Comment:
Given how many ways the config can be modified/loaded I think this is a problem. What about reading a sys.prop instead?

Reply:
I considered that as well but that doesn't seem necessarily better as it would introduce another configuration scheme. I agree with the general thought that this might get confusing when you use code-based system configuration but since this is more of a cosmetic setting, I'm not sure I clearly see the danger. In any case, I added a bit of documentation that this setting needs to be placed into an application.conf on the classpath or on the JVM command line.

Reply:
I was thinking that it is more side-effect free, sort of since this will load a potentially large set of config files in a surprising place, but perhaps it is cached so that may be ok. Still don't quite like it though, hmmmm.

Reply:
Could we instead of directly encoding it in the name string add an additional field on the name attribute and then include that in the strings where it is useful?

Reply:
> Could we instead of directly encoding it in the name string add an additional field on the name attribute and then include that in the strings where it is useful?

That's a great idea, maybe even just add another attribute with source information. The name would have automatically worked for current usages but maybe that's a shortcut we shouldn't take.